File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& v, int n) {
        vector<int> tmp;
        tmp.emplace_back(0);
        for(auto x: v){
            tmp.emplace_back(x);
        }
        tmp.emplace_back(0);
        
        for(int i=1;i<tmp.size()-1;i++){
            if(tmp[i]==0 and tmp[i-1]==0 and tmp[i+1]==0){
                tmp[i]=1;
                n-=1;
            }
        }
        return n<=0;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, lst: List[int], n: int) -> bool:
        lst = [0]+lst+[0]
        
        for i in range(1,len(lst)-1):
            if lst[i]==0 and lst[i-1]==0 and lst[i+1]==0:
                lst[i]=1
                n-=1
                
        return n<=0
        
        # _ 0 0 1 0 1 0 1 0 0 1 _
        
        [0,1,0,1,0,0,1,0]
2

[0,1,0,1,0,0,0]
1